File: HYPERLEDGER/bfn-network-apps/businesslibrary/lib/data/delivery_acceptance.dart
import 'package:businesslibrary/util/Finders.dart';

class DeliveryAcceptance extends Findable {
  String acceptanceId;
  String date;
  String deliveryNote;
  String customer, purchaseOrder, purchaseOrderNumber, customerName;
  String user, supplier;

  DeliveryAcceptance(
      {this.acceptanceId,
      this.date,
      this.deliveryNote,
      this.customer,
      this.user,
      this.supplier,
      this.customerName,
      this.purchaseOrderNumber,
      this.purchaseOrder});

  DeliveryAcceptance.fromJson(Map data) {
    this.acceptanceId = data['acceptanceId'];
    this.date = data['date'];
    this.deliveryNote = data['deliveryNote'];
    this.customer = data['customer'];
    this.user = data['user'];
    this.supplier = data['supplier'];
    this.purchaseOrder = data['purchaseOrder'];

    this.customerName = data['customerName'];
    this.purchaseOrderNumber = data['purchaseOrderNumber'];
  }
  Map<String, String> toJson() => <String, String>{
        'acceptanceId': acceptanceId == null ? ' n/a ' : acceptanceId,
        'date': date == null ? ' n/a ' : date,
        'deliveryNote': deliveryNote,
        'customer': customer,
        'purchaseOrder': purchaseOrder == null ? ' n/a ' : purchaseOrder,
        'user': user == null ? ' n/a ' : user,
        'supplier': supplier,
        'customerName': customerName,
        'purchaseOrderNumber':
            purchaseOrderNumber == null ? ' n/a ' : purchaseOrderNumber,
      };
}

File: HYPERLEDGER/bfn-network-chaincode/bfnwebapi-node/src/server/models/delivery-acceptance.ts

import { JsonObject, JsonProperty } from 'json2typescript';

@JsonObject('DeliveryAcceptance')
export class DeliveryAcceptance {
  @JsonProperty('acceptanceId', String)
  public acceptanceId: string | undefined;

  @JsonProperty('docType', String)
  public docType?: string;

  @JsonProperty('customer', String)
  public customer: string | undefined;
  @JsonProperty('customerName', String)
  public customerName: string | undefined;

  @JsonProperty('deliveryNote', String)
  public deliveryNote: string | undefined;

  @JsonProperty('supplier', String)
  public supplier: string | undefined;
  @JsonProperty('supplierName', String)
  public supplierName: string | undefined;

  @JsonProperty('purchaseOrder', String)
  public purchaseOrder: string | undefined;

  @JsonProperty('date', String)
  public date: string | undefined;
}
